ddl/partition.go

```diff
@@ -187,6 +187,10 @@ func (w *worker) onAddTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (v
 func alterTablePartitionBundles(t *meta.Meta, tblInfo *model.TableInfo, addingDefinitions []model.PartitionDefinition) ([]*placement.Bundle, error) {
 	var bundles []*placement.Bundle
 
+	// tblInfo do not include added partitions, so we should add them first
+	tblInfo = tblInfo.Clone()
+	tblInfo.Partition.Definitions = append(tblInfo.Partition.Definitions, addingDefinitions...)
```

Comment:
seems the test and code are irrelevant

Reply:
It is related in fact. The test will fail if we revert this two line. This bug only occurs when the table has its own placement settings and then we add a new partition without placement (that is to say, inherit from table). So I modified the test case to let the partition table has the placement settings in table level.